grammar.js

```diff
@@ -1,169 +1,278 @@
     simplexpr: $ =>
       choice(
-        prec.left(10, $.bool_literal),
-        prec.left(10, $.num_literal),
-        prec.left(10, $.string),
-        prec.left(10, $.ident),
-        prec.left(10, seq("(", $.simplexpr, ")")),
+        $._literal,
+        $.symbol,
         $.json_array,
         $.json_object,
-        $.function_call,
         $.json_access,
         $.json_safe_access,
         $.json_dot_access,
         $.json_safe_dot_access,
+        $.function_call,
         $.unary_expression,
         $.binary_expression,
-        $.ternary_expression
-      ),
-    ident: _ => /[a-zA-Z_][a-zA-Z0-9_-]*/,
-    json_array: $ => prec.left(10, seq("[", commaSep($.simplexpr), "]")),
-    json_object: $ =>
-      prec.left(
-        10,
-        seq("{", commaSep(seq($.simplexpr, ":", $.simplexpr)), "}")
-      ),
-    function_call: $ =>
-      prec.right(
-        9,
-        seq($.ident, field("operator", "("), commaSep($.simplexpr), ")")
+        $.ternary_expression,
+        $.parenthesized_expression,
       ),
-    json_access: $ =>
-      prec.right(
-        9,
-        seq(
-          $.simplexpr,
-          field("operator", "["),
-          alias($.simplexpr, $.index),
-          "]"
-        )
-      ),
-    json_safe_access: $ =>
-      prec.right(
-        9,
-        seq(
-          $.simplexpr,
-          field("operator", "?."),
-          "[",
-          alias($.simplexpr, $.index),
-          "]"
-        )
+
+    block: $ => seq('{', $.simplexpr, '}'),
+
+    json_array: $ => seq(
+      '[',
+      commaSep($.simplexpr),
+      ']',
+    ),
+    json_object: $ => seq(
+      '{',
+      commaSep(seq($.simplexpr, ':', $.simplexpr)),
+      '}',
+    ),
+    json_access: $ => seq(
+      $.simplexpr,
+      '[',
+      $.simplexpr,
+      ']',
+    ),
+    json_safe_access: $ => seq(
+      $.simplexpr,
+      '?.',
+      '[',
+      $.simplexpr,
+      ']',
+    ),
+    json_dot_access: $ => seq(
+      $.simplexpr,
+      '.',
+      alias($.symbol, $.index),
+    ),
+    json_safe_dot_access: $ => seq(
+      $.simplexpr,
+      '?.',
+      alias($.symbol, $.index),
+    ),
+
+    function_call: $ => seq(
+      alias($.symbol, $.function_symbol),
+      '(',
+      commaSep($.simplexpr),
+      ')',
+    ),
+
+    list: $ => seq('(', repeat($.expression), ')'),
+    array: $ => seq('[', repeat($.expression), ']'),
+
+    binary_expression: $ => {
+      const table = [
+        ['+', PREC.add],
+        ['-', PREC.add],
+        ['*', PREC.times],
+        ['/', PREC.times],
+        ['%', PREC.times],
+        ['&&', PREC.and],
+        ['||', PREC.or],
+        ['==', PREC.equality],
+        ['!=', PREC.equality],
+        ['=~', PREC.equality],
+        ['>=', PREC.relation],
+        ['<=', PREC.relation],
+        ['>', PREC.relation],
+        ['<', PREC.relation],
+        ['?:', PREC.elvis],
+      ];
+
+      return choice(...table.map(([operator, precedence]) => {
```

Comment:
I think this maintains more clarity having a simple table with the operators and precedence, vs calling a function n times. It might seem unclear at first for someone new to tree-sitter/js, but this is a very standard way of doing things